Cargo.lock

```diff
@@ -1311,7 +1345,9 @@ dependencies = [
 "checksum cloudabi 0.0.3 (registry+https://github.com/rust-lang/crates.io-index)" = "ddfc5b9aa5d4507acaf872de71051dfd0e309860e88966e1051e462a077aac4f"
 "checksum constant_time_eq 0.1.4 (registry+https://github.com/rust-lang/crates.io-index)" = "995a44c877f9212528ccc74b21a232f66ad69001e40ede5bcee2ac9ef2657120"
 "checksum crossbeam-channel 0.3.9 (registry+https://github.com/rust-lang/crates.io-index)" = "c8ec7fcd21571dc78f96cc96243cab8d8f035247c3efd16c687be154c3fa9efa"
+"checksum crossbeam-channel 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)" = "cced8691919c02aac3cb0a1bc2e9b73d89e832bf9a06fc579d4e71b68a2da061"
 "checksum crossbeam-utils 0.6.6 (registry+https://github.com/rust-lang/crates.io-index)" = "04973fa96e96579258a5091af6003abde64af786b860f18622b82e026cca60e6"
+"checksum crossbeam-utils 0.7.2 (registry+https://github.com/rust-lang/crates.io-index)" = "c3c7c73a2d1e9fc0886a08b93e98eb643461230d5f1925e4036204d5f2e261a8"
```

Comment:
Hm, now we have lots of duplicates here. Can we also update the other dependency to keep the transitive graph small?

Reply:
Unfortunately, `notify v5.0.0-pre.1` depends on `crossbeam-channel v0.3` so we can't use `crossbeam-channel v0.4.2` throughout.

We can't update `notify` easily because there are breaking changes between the version we're currently using and the next.

```
[nix-shell:~/Code/lorri]$ cargo tree -p crossbeam-utils:0.7.2 -i
crossbeam-utils v0.7.2
└── crossbeam-channel v0.4.2
    └── slog-async v2.5.0
        └── lorri v1.0.0 (/home/leo/Code/lorri)

[nix-shell:~/Code/lorri]$ cargo tree -p crossbeam-utils:0.6.6 -i
crossbeam-utils v0.6.6
└── crossbeam-channel v0.3.9
    ├── lorri v1.0.0 (/home/leo/Code/lorri)
    └── notify v5.0.0-pre.1
        └── lorri v1.0.0 (/home/leo/Code/lorri) (*)
```

However, overall this PR reduces the size of the dependencies specification by about 100 lines and removes a few transitive dependencies that are no longer required by newer versions of our dependencies (pretty amazing actually: newer versions reduce dependencies instead of adding new ones!)

Reply:
Thanks for the `cargo tree` example, I will have to remember that feature.

> pretty amazing actually: newer versions reduce dependencies instead of adding new ones!

I would sure hope the rust ecosystem settles down to a tight dependency tree with less breaking changes as time goes on, as opposed to the hot mess (high entropy!) it is right now :P

Reply:
Responding to this thread and partially to https://github.com/target/lorri/issues/351#issuecomment-603467981: when we adopted `slog`, we sort of went all-in. In this PR, I just removed `slog_async` and replaced it by a global mutex. I think that's fine because logging is unlikely to be a bottleneck. If it ever becomes one, we'll just add `slog_async` back in.

This also removes the need for a second `crossbeam-channel`:

```
[nix-shell:~/Code/lorri]$ cargo tree -p crossbeam-channel -i
crossbeam-channel v0.3.9
├── lorri v1.0.0 (/home/leo/Code/lorri)
└── notify v5.0.0-pre.1
    └── lorri v1.0.0 (/home/leo/Code/lorri) (*)
```